src/viadot/orchestration/prefect/flows/onestream_data_adapters_to_redshift_spectrum.py

```diff
@@ -0,0 +1,189 @@
+"""Extract data from OneStream Data Adapters and load it to Redshift Spectrum."""
+
+from typing import Any, Literal
+
+from prefect import flow
+from prefect.logging import get_run_logger
+
+from viadot.orchestration.prefect.tasks import (
+    create_batch_list_of_custom_subst_vars,
+    df_to_redshift_spectrum,
+    onestream_get_agg_adapter_endpoint_data_to_df,
+)
+
+
+@flow(
+    name="extract--onestream_data_adapters--redshift_spectrum",
+    description="Extract data from OneStream Data Adapters and load it to Redshift Spectrum.",
+    retries=1,
+    retry_delay_seconds=60,
+)
+def onestream_data_adapters_to_redshift_spectrum(  # noqa: PLR0913
+    server_url: str,
+    application: str,
+    adapter_name: str,
+    to_path: str,
+    schema_name: str,
+    table: str,
+    workspace_name: str = "MainWorkspace",
+    adapter_response_key: str = "Results",
+    custom_subst_vars: dict[str, list[Any]] | None = None,
+    batch_by_subst_vars: bool = False,
+    api_params: dict[str, str] | None = None,
+    extension: str = ".parquet",
+    if_exists: Literal["overwrite", "append"] = "overwrite",
+    if_empty: Literal["warn", "skip", "fail"] = "fail",
+    partition_cols: list[str] | None = None,
+    index: bool = False,
+    compression: str | None = None,
+    sep: str = ",",
+    aws_config_key: str | None = None,
+    credentials_secret: str | None = None,
+    onestream_credentials_secret: str | None = None,
+    onestream_config_key: str | None = None,
+) -> None:
+    """Extract data from OneStream Data Adapter and load it into AWS Redshift Spectrum.
+
+    This function retrieves data from a OneStream Data Adapter using provided parameters
+    and uploads it to AWS Redshift Spectrum.
+
+    When custom_subst_vars are provided and batch_by_subst_vars is True, the ingestion
+    process will be split into batches. Each batch represents one combination of the
+    custom_subst_vars. When batch_by_subst_vars is False, all substitution variable
+    combinations are processed together into a single data frame.
+    Warning! Processing custom substitution vars without batching might lead to
+    out-of-memory errors when data size is too big.
+
+    Args:
+        server_url (str): OneStream server URL.
+        application (str): OneStream application name.
+        adapter_name (str): Data Adapter name to query.
+        to_path (str): Path to an S3 folder where the table will be located.
+        schema_name (str): AWS Glue catalog database name.
+        table (str): AWS Glue catalog table name.
+        workspace_name (str): OneStream workspace name. Defaults to "MainWorkspace".
+        adapter_response_key (str): Key in the JSON response that contains
+            the adapter's returned data. Defaults to "Results".
+        custom_subst_vars (dict[str, list[Any]], optional): A dictionary mapping
```

Comment:
What are "substitution variables"? It's not self-explanatory, so can you explain here the general concept before going into the details? Maybe add an example as well if it's hard to explain?